internal/git/util.go

```diff
@@ -135,18 +135,20 @@ func isRootDir(path string) bool {
 	return len(path) == 0
 }
 
-// getLocalRepoPath will return the local path where repo will be cloned, by default it will use /tmp as base directory
+// getLocalRepoPath will return the local path where repo will be cloned,
+// by default it will use /tmp as base directory
 // unless LOCAL_REPO_PATH env is set.
-func getLocalRepoPath(gitSyncName string) string {
+func getLocalRepoPath(gitSync *v1alpha1.GitSync) string {
 	baseDir := os.Getenv("LOCAL_REPO_PATH")
 	if baseDir != "" {
-		return fmt.Sprintf("%s/%s", baseDir, gitSyncName)
+		return fmt.Sprintf("%s/%s/%s", baseDir, gitSync.Name, gitSync.Spec.RepoUrl)
 	} else {
-		return fmt.Sprintf("/tmp/%s", gitSyncName)
+		return fmt.Sprintf("/tmp/%s/%s", gitSync.Name, gitSync.Spec.RepoUrl)
```

Comment:
Are we expecting that the path will be like "/tmp/gitsync-example/https://github.com/xdevxy/numaflow-example-pipelines.git"? I'm not sure how it will be a valid one 😕

Reply:
good point, maybe we could create a hash for it instead?

Reply:
It is valid, as I test it, but a hash is better.